napari/experimental/_progressive_loading.py

```diff
@@ -0,0 +1,1652 @@
+    and will be used for calculating centers. If array is <3D, the third
+    dimension is assumed to be None.
+
+
+    Parameters
+    ----------
+    array: dask or zarr Array
+        The input array, a single scale
+
+    Returns
+    -------
+    chunk_slices: list of slice objects
+        List of slice objects for each chunk for each dimension
+    """
+    if isinstance(array, da.Array):
+        start_pos = [np.cumsum(sizes) - sizes for sizes in array.chunks]
+        end_pos = [np.cumsum(sizes) for sizes in array.chunks]
+    else:
+        # For zarr
+        start_pos = []
+        end_pos = []
+        for dim in range(len(array.chunks)):
+            # TODO: the +1 used on stop_idx is related to searchsorted usage
+            start_idx, stop_idx = 0, (array.shape[dim] + 1)
+            if interval is not None:
+                start_idx = (
+                    np.floor(interval[0, dim] / array.chunks[dim])
+                    * array.chunks[dim]
+                )
+                stop_idx = (
+                    np.ceil(interval[1, dim] / array.chunks[dim])
+                    * array.chunks[dim]
+                    + 1
+                )
+            # Inclusive on the end point
+            cumuchunks = list(
+                range(int(start_idx), int(stop_idx), array.chunks[dim])
+            )
+            cumuchunks = np.array(cumuchunks)
+            start_pos += [cumuchunks[:-1]]
+            end_pos += [cumuchunks[1:]]
+
+    if interval is not None:
+        for dim in range(len(start_pos)):
+            # Find first index in end_pos that is greater than corner_pixels
+            first_idx = np.searchsorted(end_pos[dim], interval[0, dim])
+            # Find the last index in start_pos that is less than
+            # corner_pixels[1,dim]
+            last_idx = np.searchsorted(
+                start_pos[dim], interval[1, dim], side='right'
+            )
+
+            start_pos[dim] = start_pos[dim][first_idx:last_idx]
+            end_pos[dim] = end_pos[dim][first_idx:last_idx]
+
+    chunk_slices: List[List] = [[]] * len(array.chunks)
+    for dim in range(len(array.chunks)):
+        chunk_slices[dim] = [
+            slice(st, end) for st, end in zip(start_pos[dim], end_pos[dim])
+        ]
+
+    return chunk_slices
+
+
+@thread_worker
+def render_sequence(
+    corner_pixels, visible_scales=[], data=None, ndisplay=2, camera=None
+):
+    """Generate multiscale chunk tuples from low to high resolution.
+
+    Parameters
+    ----------
+    corner_pixels : tuple
+        ND coordinates of the topleft bottomright coordinates of the
+        current view
+    full_shape : tuple
+        shape of highest resolution array
+    num_threads : int
+        number of threads for multithreaded fetching
+    visible_scales : list
+        this is used to constrain the number of scales that are rendered
+    """
+    # NOTE this corner_pixels means something else and should be renamed
```

Comment:
What do you mean by something else here? I may have misread the code, but it looks like this is `Layer.corner_pixels` from the highest resolution layer.